app/controllers/shelters_controller.rb

```diff
@@ -17,8 +17,15 @@ def create
   end
 
   def destroy
-    Shelter.destroy(params[:id])
-    redirect_to '/shelters'
+    shelter = Shelter.find(params[:id])
+    if shelter.check_applications == false
```

Comment:
I think a great way to rename this would be:
`unless shelter.applications_pending`